### PR TITLE
Using `api.StreamsOf` in discovery service

### PR DESCRIPTION
**Pull request by oxisto**
This PR removes the custom streams implementation from the discovery service and migrates it to `api.StreamsOf`, similar to the other services.

### CODE REVIEW

File: service/discovery/discovery.go [hunk 1]
```diff
@@ -100,9 +104,12 @@ const (
 type ServiceOption func(*Service)
 
 // WithAssessmentAddress is an option to configure the assessment service gRPC address.
-func WithAssessmentAddress(address string) ServiceOption {
+func WithAssessmentAddress(address string, opts ...grpc.DialOption) ServiceOption {
 	return func(s *Service) {
-		s.assessmentAddress = address
+		s.assessmentAddress = grpcTarget{
```
Comment (anatheka) [inline service/discovery/discovery.go:109]: In the assessment I have done an additional empty string check. I am not sure if it is really necessary: 
`if address == "" {
			address = DefaultAssessmentAddress
		}`

File: service/discovery/discovery.go [hunk 2]
```diff
@@ -127,10 +134,13 @@ func WithProviders(providersList []string) ServiceOption {
 
 func NewService(opts ...ServiceOption) *Service {
 	s := &Service{
-		assessmentAddress: DefaultAssessmentAddress,
-		resources:         make(map[string]voc.IsCloudResource),
-		scheduler:         gocron.NewScheduler(time.UTC),
-		configurations:    make(map[discovery.Discoverer]*Configuration),
+		assessmentStreams: api.NewStreamsOf[assessment.Assessment_AssessEvidencesClient, *assessment.AssessEvidenceRequest](),
```
Comment (anatheka) [inline service/discovery/discovery.go:137]:
```suggestion
		assessmentStreams: api.NewStreamsOf[assessment.Assessment_AssessEvidencesClient, *assessment.AssessEvidenceRequest](log),
```

File: service/discovery/discovery_test.go [hunk 3]
```diff
@@ -472,96 +445,6 @@ func TestStart(t *testing.T) {
 	}
 }
 
-func TestService_initAssessmentStream(t *testing.T) {
```
Comment (anatheka) [inline service/discovery/discovery_test.go:475]: I wouldn't delete that completely.

**Review by anatheka — APPROVED** [review]
Great! Only some minor things.

File: service/discovery/discovery.go (hunk 1 above)
Comment (oxisto) [reply service/discovery/discovery.go:109]: Should not be necessary I think

**Pull request merged**